framework/wcf/Extensibility/Transport/AzureQueueStorageClient/AzureQueueStorageClient/AzureQueueStorageChannelHelpers.cs

```diff
@@ -0,0 +1,124 @@
+    {
+        internal const string EventLogSourceName = "Microsoft.Samples.AzureQueueStorage";
+        internal const string Scheme = "soap.aqs";
+        internal const string AzureQueueStorageBindingSectionName = "system.serviceModel/bindings/sampleProfileAzureQueueStorageBinding";
+        internal const string AzureQueueStorageTransportSectionName = "aqsTransport";
+        internal const int WSAETIMEDOUT = 10060;
+
+        static MessageEncoderFactory messageEncoderFactory;
+        static AzureQueueStorageConstants()
+        {
+            messageEncoderFactory = new TextMessageEncodingBindingElement().CreateMessageEncoderFactory();
+        }
+
+        // ensure our advertised MessageVersion matches the version we're
+        // using to serialize/deserialize data to/from the wire
+        internal static MessageVersion MessageVersion
+        {
+            get
+            {
+                return messageEncoderFactory.MessageVersion;
+            }
+        }
+
+        // we can use the same encoder for all our Udp Channels as it's free-threaded
+        internal static MessageEncoderFactory DefaultMessageEncoderFactory
+       {
+            get
+            {
+                return messageEncoderFactory;
+            }
+        }
+    }
+
+    static class UdpConfigurationStrings
+    {
+        public const string MaxBufferPoolSize = "maxBufferPoolSize";
+        public const string MaxReceivedMessageSize = "maxMessageSize";
+        public const string Multicast = "multicast";
+        public const string OrderedSession = "orderedSession";
+        public const string ReliableSessionEnabled = "reliableSessionEnabled";
+        public const string SessionInactivityTimeout = "sessionInactivityTimeout";
+        public const string ClientBaseAddress = "clientBaseAddress";
+    }
+
+    static class UdpPolicyStrings
+    {
+        public const string UdpNamespace = "http://sample.schemas.microsoft.com/policy/aqs";
+        public const string Prefix = "aqs";
+        public const string MulticastAssertion = "Multicast";
+        public const string TransportAssertion = "soap.aqs";
+    }
+
+    static class AzureQueueStorageChannelHelpers
+    {
+        /// <summary>
+        /// The Channel layer normalizes exceptions thrown by the underlying networking implementations
+        /// into subclasses of CommunicationException, so that Channels can be used polymorphically from
+        /// an exception handling perspective.
+        /// </summary>
+        internal static CommunicationException ConvertTransferException(SocketException socketException)
+        {
+            return new CommunicationException(
+                string.Format(CultureInfo.CurrentCulture, 
+                "A Udp error ({0}: {1}) occurred while transmitting data.", socketException.ErrorCode, socketException.Message), 
+                socketException);
+        }
+
+        internal static bool IsInMulticastRange(IPAddress address)
+        {
```

Comment:
Not needed for queues, this is specific to udp

Reply:
got it, removing this.